src/main/java/teammates/ui/webapi/action/BinCourseAction.java

```diff
@@ -0,0 +1,43 @@
+package teammates.ui.webapi.action;
+
+import org.apache.http.HttpStatus;
+
+import teammates.common.exception.EntityDoesNotExistException;
+import teammates.common.exception.UnauthorizedAccessException;
+import teammates.common.util.Const;
+
+/**
+ * Soft delete a course.
```

Comment:
I prefer not to expose the term `soft-delete` in the application level. We should mention `Move a course to to recycle bin instead`.